java/ruleXtract/src/main/java/uk/ac/cam/eng/extraction/hadoop/features/phrase/PhraseJob.java

```diff
@@ -0,0 +1,60 @@
+/*******************************************************************************
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use these files except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ *******************************************************************************/
+
+package uk.ac.cam.eng.extraction.hadoop.features.phrase;
+
+import java.io.IOException;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.conf.Configured;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.mapreduce.Job;
+import org.apache.hadoop.mapreduce.lib.input.FileInputFormat;
+import org.apache.hadoop.mapreduce.lib.output.FileOutputFormat;
+import org.apache.hadoop.util.Tool;
+
+import uk.ac.cam.eng.extraction.hadoop.util.Util;
+import uk.ac.cam.eng.util.CLI;
+
+import com.beust.jcommander.ParameterException;
+
+/**
+ * Common functionality between the phrase (s2t, t2s) jobs
+ * @author Aurelien Waite
+ *
+ */
+
+public abstract class PhraseJob extends Configured implements Tool {
+
+	public abstract Job getJob(Configuration conf) throws IOException;
+	
+	@Override
+	public int run(String[] args) throws IllegalArgumentException,
+	IllegalAccessException, IOException, ClassNotFoundException,
+	InterruptedException {
+		CLI.MarginalReducerParameters params = new CLI.MarginalReducerParameters();
+		try {
+			Util.parseCommandLine(args, params);
+		} catch (ParameterException e) {
+			return 1;
+		}
+		Configuration conf = getConf();
+		Util.ApplyConf(params, conf);
+		Job job = getJob(conf);
+		FileInputFormat.setInputPaths(job, params.input);
+		FileOutputFormat.setOutputPath(job, new Path(params.output));
+		return job.waitForCompletion(true) ? 0 : 1;
```

Comment:
These 0,1 values seem to be the same return values used in other places. You could specify constants instead.